src/pk.c

```diff
@@ -7433,6 +7433,10 @@ static WOLFSSL_DH *wolfssl_dhparams_read_pem(WOLFSSL_DH **dh,
                 != 0) {
             err = 1;
         }
+        /* If Success on X9.42 DH format, clear error from failed DH format */
+        else {
+            wolfSSL_ERR_clear_error();
```

Comment:
wolfSSL_ERR_clear_error() calls wc_ClearErrorNodes which clears out the whole error queue. Is there a way that you could pop the most recent error here (and sanity check if is from failed DH format)? Clearing the whole queue could clear out error nodes that are needed elsewhere.

Reply:
Fixed